mas-connecta/src/main/java/com/ca/mas/connecta/client/MASConnectaManager.java

```diff
@@ -107,32 +65,88 @@ public void setConnectaListener(MASConnectaListener listener) {
         connectaListener = listener;
     }
 
+    /**
+     * @deprecated Use {@link MASConnectaManager#disconnect(MASCallback)}
+     */
+    @Deprecated
     public void stop() {
         disconnect(null);
     }
 
+    private MqttConnecta getMqttConnecta() {
+        //If no connect Options is provided, default to the gateway
+        if ((connectOptions == null) || (connectOptions.getServerURIs() == null)) {
+            return new GatewayMqttConnecta();
+        } else {
+            return new PublicMqttConnecta((clientId));
+        }
+    }
+
     @Override
-    public synchronized void connect(MASCallback<Void> callback) {
+    public synchronized void connect(final MASCallback<Void> callback) {
+
         if (isConnected()) {
             Callback.onSuccess(callback, null);
             return;
         }
-        if (mMASTransportService == null) {
-            connectCallback = callback;
-            Intent intent = new Intent(MAS.getContext(), ConnectaService.class);
-            MAS.getContext().bindService(intent, mServiceConnection, Context.BIND_AUTO_CREATE);
-        } else {
-            mMASTransportService.connect(callback);
+        final MqttConnecta mqttConnecta = getMqttConnecta();
+
+        if (connectOptions == null) {
+            connectOptions = new MASConnectOptions();
         }
+        if (timeOutInMillis > 0) {
```

Comment:
Is there any need to keep these two conditions separate and not merged in the same line?